### PR TITLE
Add deterministic support in nn.functional.interpolate for XPU

### PR DESCRIPTION
**Pull request by majing921201**
Both for CUDA and XPU, there are no deterministic implementation at native in `aten::upsample_bilinear` and `aten::replication_pad`. CUDA leverage operator decomposition path in frontend hook `nn.functional.interpolate` as its deterministic implentation. XPU backend uses the same solution in this PR.

cc @frank-wei @jgong5 @mingfeima @XiaobingSuper @sanchitintel @ashokei @jingxu10 @gujinghui @EikanWang @fengyuan14 @guangyey

### CODE REVIEW

File: torch/nn/functional.py
Comment (dvrogozh) [inline torch/nn/functional.py:4571]: 1. Is this check needed at the first place? I mean I wonder whether `input.is_cuda()` condition should be just removed instead?
2. If it's really needed, then why this check is added here and is not added here (this 2nd place is in `pad()`):
https://github.com/pytorch/pytorch/blob/0d0305e774fc40833047bf95fbb49f1a35615f2a/torch/nn/functional.py#L5084

Comment (majing921201) [reply torch/nn/functional.py:4571]: > 1. Is this check needed at the first place? I mean I wonder whether `input.is_cuda()` condition should be just removed instead?
> 2. If it's really needed, then why this check is added here and is not added here (this 2nd place is in `pad()`):
>    https://github.com/pytorch/pytorch/blob/0d0305e774fc40833047bf95fbb49f1a35615f2a/torch/nn/functional.py#L5084

1, For other backend, like cpu, whose kernel is deterministic implementation. So there is not need to call decompose implementation, which has slow performance.
2, yes, we also need the 2nd, I am working on interpolate, and haven't validate pad related functionality. Would you like a sperate pr or, I just add pad change in this pr ?

Comment (dvrogozh) [reply torch/nn/functional.py:4571]: > Would you like a sperate pr or, I just add pad change in this pr?

I suggest to have this in this PR.

File: torch/nn/functional.py
Comment (dvrogozh) [inline torch/nn/functional.py:4572]: >> Is this check needed at the first place? I mean I wonder whether input.is_cuda() condition should be just removed instead?
>
> For other backend, like cpu, whose kernel is deterministic implementation. So there is not need to call decompose implementation, which has slow performance.

I still think it's better to have some other check here instead of explicit iterations over specific backends. Dedicated backend attribute to check might be a better variant. However, that's something to decide on by pytorch maintainers. So, I'll approve the PR from my side and leave that to maintainers.

**Review by dvrogozh — APPROVED** [review]

**Comment by EikanWang** [comment]
Please check the CI failures.

**Comment by EikanWang** [comment]
By the way, please refine the PR description.

**Comment by majing921201** [comment]
> Please check the CI failures.

I have no idea for the failures in inductor/test_torchinductor.py::GPUTests::test_convolution5_xpu. The test case looks like only have convolution operator, will check offline.

**Comment by EikanWang** [comment]
@albanD , @atalman , may I know if you have any comments on this PR?

**Review by albanD — APPROVED** [review]
Sounds ok. While I agree with @dvrogozh I don't want to go too far in the backend flag side unless we have proper scaffolding to ensure we don't break out of tree backends and we can easily add/remove flags in a BC way.

**Review by EikanWang — APPROVED** [review]

**Pull request closed without merging**
